rust/rust.bzl

```diff
@@ -156,7 +169,7 @@ def _setup_deps(deps, name, working_dir, allow_cc_deps=False,
       symlinked_libs += [dep.rust_lib] + dep.transitive_libs
       link_flags += [(
           "--extern " + dep.label.name + "=" +
-          deps_dir + "/" + dep.rust_lib.basename
+          deps_dir + "/" + _hashed_lib_path(dep.rust_lib)
```

Comment:
Can we hash the libname in the output instead? ie. rust_library.outputs = _hash_lib_name

I'm not clear on whether the callback api makes the path available, but it would simplify this logic a bit.

Reply:
Can you elaborate a bit on this suggestion? I think I have an understanding: You'd like to see the output (rlib|so|...) themselves be uniqueified, rather than doing this at link time?

Reply:
The way this is written rust_library.rust_lib is only ever consumed here, and it's incorrect to depend on the non-hashed rlib, so it's better to not expose the non-hashed rlib.

Somewhat premature, but this also repeats the _hashed_lib_path function in each dependent instead of once in the library.

(iirc .so and .a should not be hashed, since their symbols could conflict)

Reply:
L168 and L169 should also be referencing the hashed path (for compiler inputs to be 100% correct), and then you don't need to modify _setup_cmd either.

Reply:
OK: Done. Please take a look